modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java

```diff
@@ -192,23 +201,45 @@ public void onReceived(NetworkMessage message, NetworkAddress senderAddr, @Nulla
 
                     ReplicaRequest request = (ReplicaRequest) message;
 
+                    HybridTimestamp requestTimestamp;
```

Comment:
Let's hide all that semantically consistent things into corresponding private methods, including response preparation even error one. All in all I'd expect something similar too
```
    public void start() {
        clusterNetSvc.messagingService().addMessageHandler(ReplicaMessageGroup.class, new NetworkMessageHandler() {
            @Override
            public void onReceived(NetworkMessage message, NetworkAddress senderAddr, @Nullable Long correlationId) {
                if (!busyLock.enterBusy()) {
                    throw new IgniteException(new NodeStoppingException());
                }

                try {
                    assert message instanceof ReplicaRequest : IgniteStringFormatter.format("Unexpected message [message={}]", message);

                    ReplicaRequest request = (ReplicaRequest) message;

                    HybridTimestamp requestTimestamp = extractTimestamp(request);

                    Replica replica = replicas.get(request.groupId());

                    if (replica == null) {
                        prepareReplicaUnavailableErrorResponse(senderAddr, correlationId, request, requestTimestamp);
                    }

                    CompletableFuture<Object> result = replica.processRequest(request);

                    result.handle((res, ex) -> {
                        NetworkMessage msg;

                        if (ex == null) {
                            msg = prepareReplicaResponse(requestTimestamp, res);
                        } else {
                            msg = prepareReplicaErrorResponse(request, requestTimestamp, ex);
                        }

                        clusterNetSvc.messagingService().respond(senderAddr, msg, correlationId);

                        return null;
                    });
                } finally {
                    busyLock.leaveBusy();
                }
            }
        });
    }
```

Reply:
Fixed.